CalibTracker/SiStripChannelGain/interface/SiStripGainsPCLWorker.h

```diff
@@ -93,15 +93,22 @@ class SiStripGainsPCLWorker : public  DQMEDAnalyzer {
 
     int statCollectionFromMode(const char* tag) const;
 
-    std::vector<MonitorElement*>  Charge_Vs_Index;
-    std::vector<MonitorElement*>  Charge_Vs_PathlengthTIB;
-    std::vector<MonitorElement*>  Charge_Vs_PathlengthTOB;
-    std::vector<MonitorElement*>  Charge_Vs_PathlengthTIDP;
-    std::vector<MonitorElement*>  Charge_Vs_PathlengthTIDM;
-    std::vector<MonitorElement*>  Charge_Vs_PathlengthTECP1;
-    std::vector<MonitorElement*>  Charge_Vs_PathlengthTECP2;
-    std::vector<MonitorElement*>  Charge_Vs_PathlengthTECM1;
-    std::vector<MonitorElement*>  Charge_Vs_PathlengthTECM2;
+    std::vector<MonitorElement*>  Charge_Vs_Index;           /*!< Charge per cm for each detector id */
+    std::vector< std::vector<MonitorElement*> > Charge_1;    /*!< Charge per cm per layer / wheel */
```

Comment:
@davidlange6 sorry I need a clarification: do you mean to change

std::vector< std::vector<MonitorElement*> >   into  std::vector<std::array<MonitorElement*,7>>  ?

This may be better but it will invert the index loop: in the former the first index goes up to 7, while the second (e.g. the internal) index is determined by the number of Charge Monitor histograms defined as input parameters; in the latter (e.g. your proposal) the first index is determined by the number of Charge Monitor histograms while the second is ranging up to 7. This implies some changes in the filling logic too, so I want to be sure I do understand you before doing it.